### PR TITLE
otel log exporter

### PR DESCRIPTION
**Pull request by phiSgr**

### CODE REVIEW

File: apps/framework-cli/src/cli/logger.rs [hunk 1]
```diff
@@ -151,6 +164,44 @@ pub fn setup_logging(settings: &LoggerSettings) -> Result<(), fern::InitError> {
...
+                .with_timeout(Duration::from_millis(10000))
+                .build_log_exporter()
+                .unwrap();
+
+            let logger_provider =
+                LoggerProvider::builder()
+                    .with_config(Config::default().with_resource(Resource::new(vec![
+                        KeyValue::new(SERVICE_NAME, "moose-cli"),
+                    ])))
+                    .with_batch_exporter(otel_exporter, opentelemetry_sdk::runtime::Tokio)
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/logger.rs]: `with_simple_exporter` gave me a dead lock when there's concurrency.
i had to do a thread dump inside a linux container to understand what went wrong.

**Comment by phiSgr** [comment]
The data sent to the endpoint looks like this:

```json
{
  "resourceLogs": [
    {
      "resource": {
        "attributes": [
          {
            "key": "service.name",
            "value": {
              "stringValue": "moose-cli"
            }
          }
        ],
        "droppedAttributesCount": 0
      },
      "scopeLogs": [
        {
          "scope": {
            "name": "opentelemetry-log-appender",
            "version": "0.4.0",
            "attributes": [],
            "droppedAttributesCount": 0
          },
          "logRecords": [
            {
              "timeUnixNano": "0",
              "observedTimeUnixNano": "1718339383881419000",
              "severityNumber": 5,
              "severityText": "DEBUG",
              "body": {
                "value": {
                  "stringValue": "[2024-06-14T04:29:43Z DEBUG 22d05d56-2c77-4524-aa8c-5e8092ee6f96 - moose_cli::infrastructure::kafka_clickhouse_sync] Error receiving message from Stuff_0_0: Message consumption error: BrokerTransportFailure (Local: Broker transport failure)"
                }
              },
              "attributes": [],
              "droppedAttributesCount": 0,
              "flags": 0,
              "traceId": "",
              "spanId": ""
            }
          ],
          "schemaUrl": ""
        }
      ],
      "schemaUrl": ""
    },
    {
      "resource": {
        "attributes": [
          {
            "key": "service.name",
            "value": {
              "stringValue": "moose-cli"
            }
          }
        ],
        "droppedAttributesCount": 0
      },
      "scopeLogs": [
        {
          "scope": {
            "name": "opentelemetry-log-appender",
            "version": "0.4.0",
            "attributes": [],
            "droppedAttributesCount": 0
          },
          "logRecords": [
            {
              "timeUnixNano": "0",
              "observedTimeUnixNano": "1718339384514454000",
              "severityNumber": 5,
              "severityText": "DEBUG",
              "body": {
                "value": {
                  "stringValue": "[2024-06-14T04:29:44Z DEBUG 22d05d56-2c77-4524-aa8c-5e8092ee6f96 - moose_cli::infrastructure::kafka_clickhouse_sync] Error receiving message from Stuff_0_0: Message consumption error: BrokerTransportFailure (Local: Broker transport failure)"
                }
              },
              "attributes": [],
              "droppedAttributesCount": 0,
              "flags": 0,
              "traceId": "",
              "spanId": ""
            }
          ],
          "schemaUrl": ""
        }
      ],
      "schemaUrl": ""
    },
    {
      "resource": {
        "attributes": [
          {
            "key": "service.name",
            "value": {
              "stringValue": "moose-cli"
            }
          }
        ],
        "droppedAttributesCount": 0
      },
      "scopeLogs": [
        {
          "scope": {
            "name": "opentelemetry-log-appender",
            "version": "0.4.0",
            "attributes": [],
            "droppedAttributesCount": 0
          },
          "logRecords": [
            {
              "timeUnixNano": "0",
              "observedTimeUnixNano": "1718339384756210000",
              "severityNumber": 5,
              "severityText": "DEBUG",
              "body": {
                "value": {
                  "stringValue": "[2024-06-14T04:29:44Z DEBUG 22d05d56-2c77-4524-aa8c-5e8092ee6f96 - moose_cli::infrastructure::kafka_clickhouse_sync] Error receiving message from Stuff_0_0: Message consumption error: BrokerTransportFailure (Local: Broker transport failure)"
                }
              },
              "attributes": [],
              "droppedAttributesCount": 0,
              "flags": 0,
              "traceId": "",
              "spanId": ""
            }
          ],
          "schemaUrl": ""
        }
      ],
      "schemaUrl": ""
    }
  ]
}
```

**Comment by callicles** [comment]
Should we try to have the log more structured?

```
"stringValue": "[2024-06-14T04:29:44Z DEBUG 22d05d56-2c77-4524-aa8c-5e8092ee6f96 - moose_cli::infrastructure::kafka_clickhouse_sync] Error receiving message from Stuff_0_0: Message consumption error: BrokerTransportFailure (Local: Broker transport failure)"
```

Looks like in the above the we could have `timestamp` removed, `level` in a property, `module` in a property and `sequenceId` in a property, no?

**Comment by phiSgr** [comment]
`timestamp` removed
`level` is `severityText`
`module` moved to `attributes` in `logRecords`
`session_id` moved to `attributes` in `resource`

**Review by callicles — APPROVED** [review]

**Pull request merged**